### DISCUSSION TITLE
Can't install necessary dotnet workloads

### DISCUSSION THREAD

**Discussion by Eatham532** [unoplatform/uno#12306] [Q&A] [answered]
Sorry in advance if this is the wrong place to ask this question.

When running the command `dotnet workload install` to download the necessary workloads for uno I get installation errors. Here is a log from powershell.  

[uno-log.txt](https://github.com/unoplatform/uno/files/11460495/uno-log.txt)

When manually running `dotnet workload install macos`  I get this warning at the end.
```Warning: Workload garbage collection failed with error: Object reference not set to an instance of an object..```

When manually running `dotnet workload install maccatylist` it begins to rollback the action. With the same error `Object reference not set to an instance of an object`

**Comment by Youssef1313** [comment]
Try https://github.com/unoplatform/uno.check/issues/50#issuecomment-1178452596

**Comment by Eatham532** [comment]
I did that before. Didn't seem to fix it.  But I will try to do it again.

**Comment by jeromelaban** [comment]
@Eatham532 thanks for the update. Why do you need to run `dotnet workloads restore` ? Does uno-check indicate that all is installed?

**Comment by Eatham532** [comment]
uno-check says that my workloads are not installed. And when trying to build the default template app I get errors about the workloads. So I tried installing them manually using `dotnet workloads install` which failed.

**Comment by Eatham532** [comment]
Could it be because I am using .net 7?

**Comment by nickrandolph** [comment]
@Eatham532 do you have any preview versions of Visual Studio installed? Also, can you run "dotnet --info" and share results here

**Comment by Eatham532** [comment]
No, I do not have any preview versions installed. The results from running `dotnet --info`:

```.NET SDK:
 Version:   7.0.203
 Commit:    5b005c19f5

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22621
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.203\

Host:
  Version:      7.0.5
  Architecture: x64
  Commit:       8042d61b17

.NET SDKs installed:
  7.0.203 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download```

**Comment by jeromelaban** [comment]
This is unfortunately a case of workloads not working properly. Aside from uninstalling all .NET versions and all VS instances (that is until .NET 8 with the new `dotnet workload cleanup` option that may help) and reinstalling everything, we cannot provide much more options.

This is a topic that is very frequent on the MAUI repo: https://github.com/dotnet/maui/issues?q=is%3Aissue+is%3Aopen+workload

<blockquote><img src="https://repository-images.githubusercontent.com/262395224/cd3ce980-99ab-11ea-800c-e4367bfd3c2d" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/dotnet/maui">Issues · dotnet/maui</a></strong></div><div>.NET MAUI is the .NET Multi-platform App UI, a framework for building native device applications spanning mobile, tablet, and desktop. - Issues · dotnet/maui</div></blockquote>

**Comment by Eatham532** [comment, marked as answer]
I managed to fix this issue. There seemed to be some issue with the workloads I currently had installed. And it seems when removing dotnet and visual studio the workloads don't get removed. I created a backup of `C:\ProgramData\dotnet\workloads` and removed the folder. Then when running `uno-check` the workloads installed successfully.

I also moved back the workloads I made a backup with before in case there were some that were not installed when running `uno-check` and made sure to click skip when there were two of the same files.

**Reply by jeromelaban** [reply]
Thanks for the update!

**Reply by Mertsch** [reply]
The issue still exists in 2025 and yes, deleting `C:\ProgramData\dotnet\workloads` and running `dotnet workload update` fixes the issue.
